src/Moat/Pretty/Swift.hs

```diff
@@ -268,3 +268,8 @@ prettyPrivateTypes indents = go
 onLast :: (a -> a) -> [a] -> [a]
 onLast _ [] = []
 onLast f (x : xs) = x : map f xs
+
+addTyVarBounds :: [String] -> [Protocol] -> [String]
+addTyVarBounds tyVars protos
+  | Codable `elem` protos = map (++ ": Codable") tyVars
+  | otherwise = tyVars
```

Comment:
Two questions:
- should this include more cases? Or is it only relevant for Codable instances
- should we make `protocolToString` to avoid the literal `"Codable"`

Reply:
Looks like we also need this for `Hashable`. I wonder if a setting like `structTyVarProtocols` and `newtypeTyVarProtocols` would make sense.

Second point, we could pull out `prettyProtocol` and use it here.

Reply:
Okay, this is only needed for "protocols with synthesized implementations" as seen here: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/protocols#Adopting-a-Protocol-Using-a-Synthesized-Implementation

So this should be hardcoded for now.